torch/_functorch/_aot_autograd/subclass_utils.py

```diff
@@ -106,28 +106,53 @@ def create_subclass_meta(
 # a list of tensors that we would then need to concat together.
 # Instead, we specialize the logic for the inference vs. joint graph case.
 # NOTE: this function is hot, since we unwrap tensor subclass inputs at runtime
-def unwrap_tensor_subclasses(wrapped_args, *, is_joint_structure: bool):
+def unwrap_tensor_subclasses(
+    wrapped_args, *, is_joint_structure: bool, return_indices: bool = False
```

Comment:
A few reasons I don't love this function (and really want to avoid making it more complicated 😛 )

(1) this function is already used in so many different contexts that it's a pain to reason about:

* unwrapping subclass example inputs at compile time before starting tracing (both for the joint graph and inference)
* unwrapping subclass graph outputs at the end of tracing
* unwrapping subclass fw inputs at runtime
* unwrapping subclass bw inputs at runtime

(2) We're also unconditionally computing the `index_remap` below even with the flag is false, which could hurt runtime performance (this fn is part of the hot-path when we invoke it at runtime to desugar subclass fw/bw inputs).

(3) you can pass in the combo of args `(is_joint_structure=True, return_indices=True)`, but we never actually need to compute the indices of tangents right now

(4) @guilhermeleobas is making this function more complicated too as part of https://github.com/pytorch/pytorch/pull/125941/files

Instead, could we define a new function like `def map_subclass_to_plain_tensor_indices(args: List[Tensor]) -> List[int]`? (or whatever form the indices mapping takes)

Reply:
Yeah I can do a separate function here, was just trying to help perf by reusing the existing iterations that we already do.

Reply:
lol this is a lot simpler